### DISCUSSION TITLE
doubt on message passing layer

### DISCUSSION THREAD

**Discussion by steffyalbert** [pyg-team/pytorch_geometric#3030] [Q&A]
I am working a particle-based fluid simulation data and  I use GNN to predict the velocity components of a particular time step given the past 4 timesteps position and velocity as input.  My final goal is to generate a rollout of the predictions of test data(giving the predicted timestep  as input and predicting the next timestep and so on). Initially, I used the GCNConv layer for the message-passing step, but  I was not able to get rollout predictions, but then when I changed the message-passing layer to EdgeConv it was able to generate rollout up to few timesteps. As both, the classes aggregate the neighborhood features what could be the possible reason for this behavior? Also is there any better implementation I could use from the Pytorch geometric package that I could use for this type of scenario?

**Comment by rusty1s** [comment]
The `GCNConv` is somewhat restricted/non-expressive, and is therefore only well-suited for learning on small amount of data. There exists numerous reasons why `EdgeConv` performs better. One reason is that `EdgeConv` performs an anisotropic aggregation, in which neighbors are transformed differently, while `GCNConv` transforms all neighbors equally (and therefore simply smooths features across local neighborhoods). I think `EdgeConv` is a good operator for your use-case. Dependent on if you have additional edge features, other operators might be worth looking into as well, such as `SplineConv` or `NNConv`.

**Reply by steffyalbert** [reply]
Thanks a lot for the information and suggestion.
